test/t8_geometry/t8_geometry_implementations/t8_gtest_geometry_occ.cxx

```diff
@@ -364,3 +366,105 @@ TEST (t8_gtest_geometry_occ, jacobian)
   t8_cmesh_destroy (&cmesh);
 }
 #endif /* T8_WITH_OCC */
+
+#if T8_WITH_OCC
+/* The test checks if the mapping algorithms for curved 2d elements do not shift values on a surface which is not curved.
+ * In that case, the occ geometry should output the same out_coords as the linear geometry function. */
+class class_2d_element_linear_occ_surface: public testing::TestWithParam<t8_eclass_t> {
+ protected:
+  void
+  SetUp () override
+  {
+    eclass = GetParam ();
+    T8_ASSERT (0 <= eclass && eclass < T8_ECLASS_COUNT);
+    Handle_Geom_Surface occ_surface;
+    TColgp_Array2OfPnt point_array (1, 2, 1, 2);
+    TopoDS_Shape shape;
+
+    /*  x--> u-parameter
+    *  |
+    *  v v-parameter
+    *
+    *     point_array  1               2
+    *
+    *         1        -----------------
+    *                  |               |
+    *                  |               |
+    *                  | plane surface |
+    *                  |               |
+    *                  |               |
+    *         2        -----------------
+    */
+
+    point_array (1, 1) = gp_Pnt (0.0, 1.0, 0.0);
+    point_array (2, 1) = gp_Pnt (1.0, 1.0, 0.0);
+
+    point_array (1, 2) = gp_Pnt (0.0, 0.0, 0.0);
+    point_array (2, 2) = gp_Pnt (1.0, 0.0, 0.0);
+
+    occ_surface = GeomAPI_PointsToBSplineSurface (point_array).Surface ();
+    shape = BRepBuilderAPI_MakeFace (occ_surface, 1e-6).Face ();
+
+    geometry_occ = new t8_geometry_occ (2, shape, "occ surface dim=2");
+
+    t8_cmesh_init (&cmesh);
+    t8_cmesh_set_tree_class (cmesh, 0, eclass);
+  }
+  void
+  TearDown () override
+  {
+    t8_cmesh_destroy (&cmesh);
+  }
+  t8_cmesh_t cmesh;
+  t8_eclass_t eclass;
+  t8_geometry_occ *geometry_occ;
+  /* The arrays prescribe the linkage of the element. The face of the element is linked and all edges are not */
+  int faces[1] = { 1 };
+  int edges[8] = { 0 };
+};
+
+TEST_P (class_2d_element_linear_occ_surface, t8_check_2d_element_linear_occ_surface)
+{
+  double vertices_quad[12] = { 0.0, 0.0, 0.0, 1.0, 0.0, 0.0, 0.0, 1.0, 0.0, 1.0, 1.0, 0.0 };
```

Comment:
I think for these arrays there are already predefinitions: https://github.com/DLR-AMR/t8code/blob/b062c4b828e0264cea3e6210a75eef6b9edf165c/src/t8_element.c#L27

You should use this instead of defining your own.